src/block.rs

```diff
@@ -97,13 +107,13 @@ impl BlockBuilder {
     #[must_use]
     pub fn add(&mut self, key: &[u8], value: Option<&[u8]>) -> bool {
         assert!(!key.is_empty(), "key must not be empty");
-        let overlap_len = compute_prefix(&self.first_key, key);
-        let rest_key = &key[overlap_len..];
+        let key_prefix_len = compute_prefix(&self.first_key, key);
+        let key_suffix = &key[key_prefix_len..];
```

Comment:
renamed these since I struggled to understand what "overlap" and "rest" meant the first time around